Actions/Github-Helper.psm1

```diff
@@ -708,87 +708,234 @@ function Set-JsonContentLF {
     }
 }
 
-function GetArtifacts {
+<#
+    Gets the last successful CICD run ID for the specified repository and branch.
+
+    If no successful CICD run is found, 0 is returned.
+#>
+
+function FindLatestSuccessfulCICDRun {
     Param(
+        [Parameter(Mandatory = $true)]
         [string] $token,
-        [string] $api_url = $ENV:GITHUB_API_URL,
-        [string] $repository = $ENV:GITHUB_REPOSITORY,
-        [string] $mask = "Apps",
-        [string] $branch = "main",
-        [string] $projects,
+        [Parameter(Mandatory = $true)]
+        [string] $api_url,
+        [Parameter(Mandatory = $true)]
+        [string] $repository,
+        [Parameter(Mandatory = $true)]
+        [string] $branch
+    )
+
+    $headers = GetHeader -token $token
+    $lastSuccessfulCICDRun = 0
+    $per_page = 100
+    $page = 1
+
+    # Get the latest CICD workflow run
+    while($true) {
+        $runsURI = "$api_url/repos/$repository/actions/runs?per_page=$per_page&page=$page&exclude_pull_requests=true&status=success&branch=$branch"
+        Write-Host "- $runsURI"
+        $workflowRuns = InvokeWebRequest -Headers $headers -Uri $runsURI | ConvertFrom-Json
+
+        if($workflowRuns.workflow_runs.Count -eq 0) {
+            # No more workflow runs, breaking out of the loop
+            break
+        }
+
+        $CICDRuns = @($workflowRuns.workflow_runs | Where-Object { $_.name -eq ' CI/CD' })
+
+        if ($CICDRuns.Count -gt 0) {
+            $lastSuccessfulCICDRun = $CICDRuns[0].id
```

Comment:
I guess this assumes that CICDRuns is sorted with the latest CICD run first? I don't see any mention of it being sorted in the documentation though (https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository). Shouldn't we be a bit cautious assuming this if it's not actually documented?

Reply:
Such things are never really documented. In all my testing, the runs were always sorted.

Ensuring sorted result will be too expensive as the function will have to query for _all_ CICD runs every time.